examples/cloudml-churn-prediction/preprocessor/preprocessor/preprocess.py

```diff
@@ -0,0 +1,342 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Build preprocessing pipeline for survival analysis."""
+import sys
+
+from . import features
+from . import query
+import apache_beam as beam
+from apache_beam.options import pipeline_options
+import argparse
+import datetime
+import numpy as np
+import os
+import random
+from tensorflow_transform import coders
+from tensorflow_transform.beam import impl as tft_beam
+from tensorflow_transform.beam import tft_beam_io
+
+
+def _random_start_date(start_interval, end_interval):
+    """Generate a random date between some time interval.
+
+    Used to randomly generate (fake) subscription start date
+    """
+
+    delta = end_interval - start_interval
+    int_delta = delta.days
+    random_day = random.randrange(int_delta)
+    return start_interval + datetime.timedelta(days=random_day)
+
+
+def _random_end_date(start):
+    """Generate a random duration and return subscription end date.
+
+    Used to random generate (fake) subscription duration, which directly relates
+    to the artificial label
+    """
+
+    random_duration = random.randrange(np.max(
+        features.LABEL_CEILINGS) + np.min(features.LABEL_CEILINGS))
+    return start + datetime.timedelta(days=random_duration)
+
+
+def _generate_fake_data(element):
+    """Appends randomly generated labels to each sample's dictionary.
+
+    This transformation is not necessory when using real data, because these
+    values should already be available and these are the labels for the
+    supervised classification churn model.
+
+    Args:
+        element: dictionary of results from BigQuery
+    Returns:
+        dictionary of results from BigQuery with the following fields added:
+            start_date: datetime object representing date when subscription
+                began.
+            end_date: datetime object representing date when subscription
+                ended. None if subscription has not yet ended.
+            active: True if the subscription is still active.
+    """
+
+    d1 = datetime.datetime.strptime('1/1/2018', '%m/%d/%Y')
+    d2 = datetime.datetime.strptime('12/25/2018', '%m/%d/%Y')
+    start_date = _random_start_date(d1, d2)
+    end_date = _random_end_date(start_date)
+    active = True if end_date > d2 else False
+    element.update(start_date=start_date)
+    if not active:
+        element.update(end_date=end_date)
+    else:
+        element.update(end_date=None)
+    element.update(active=active)
+    return element
+
+
+def _map_to_class(element):
```

Comment:
nit: name is misleading. 
consider: s/_map_to_class/_enhance_with_class_and_duration/ or similar

Reply:
This should be unittested.
This function does two things, consider splitting up.

Reply:
Almost done adding unit testing. Should update PR tomorrow